tiledb/sm/array_schema/dimension.h

```diff
@@ -821,12 +821,29 @@ class Dimension {
   /**
    * If the tile extent is `null`, this function sets the
    * the tile extent to the dimension domain range.
+   * Applicable to integral domains.
    *
    * @tparam T The dimension type.
    *
    * @note This is applicable only to dense arrays.
    */
-  template <class T>
+  template <
+      class T,
+      typename std::enable_if<std::is_integral<T>::value>::type* = nullptr>
+  Status set_null_tile_extent_to_range();
+
+  /**
+   * If the tile extent is `null`, this function sets the
+   * the tile extent to the dimension domain range.
+   * Applicable to real domains.
+   *
+   * @tparam T The dimension type.
+   *
+   * @note This is applicable only to dense arrays.
+   */
+  template <
+      class T,
+      typename std::enable_if<!std::is_integral<T>::value>::type* = nullptr>
```

Comment:
We're only instantiating this function for `float` and `double`; there are more non-integral domains than floating point ones. The template condition is for non-integral types. It might be desirable change the template condition to `std::is_floating_point`.